### PR TITLE
Document the system audit-log #209

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Review by rymsha — APPROVED** [review]

File: docs/auditLogs.adoc [hunk 1]
```diff
@@ -0,0 +1,150 @@
+= Audit Logs
+:toc: right
+
+XP registers only successfully completed actions in the `system.auditlog` repository for `content`, `applications`, `principals` and `ID providers`.
```
Comment (alansemenov) [inline docs/auditLogs.adoc:4]:
```suggestion
XP registers successfully completed actions that affect content, applications, principals and ID providers in the `system.auditlog` repository.
```

File: docs/auditLogs.adoc [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+* `system.security.principal.delete` - a principal (user, group or role) has been deleted
+* `system.security.principal.addRelationship` - a principal joins a group or a role
+* `system.security.principal.removeRelationship` - a principal leaves a group or a role
+* `system.security.principal.removeRelationships` - all group or role relationships have been removed (deprecated)
+* `system.security.principal.setPassword` - new password has been set for a user
+* `system.security.idProvider.create` - an ID provider has been created
+* `system.security.idProvider.update` - an ID provider has been updated
+* `system.security.idProvider.delete` - an ID provider has been deleted
+
+Below you can see the examples how data stores for actions with the following types:
```
Comment (alansemenov) [inline docs/auditLogs.adoc:30]:
```suggestion
Below you can see some examples of how data is stored for various actions:
```

File: docs/auditLogs.adoc [hunk 3]
```diff
@@ -0,0 +1,150 @@
+= Audit Logs
+:toc: right
+
+XP registers only successfully completed actions in the `system.auditlog` repository for `content`, `applications`, `principals` and `ID providers`.
+
+== Principals and ID providers
+
+In order to enable or disable audit logs for `ID providers` and principals entities such as `users`, `roles` and `groups` you have to reference to the `com.enonic.xp.security.cfg` configuration file and set value as `true` to enable and as `false` to disable for the `auditlog.enabled` option. Default value is `true`.
```
Comment (alansemenov) [inline docs/auditLogs.adoc:8]:
```suggestion
In order to enable or disable audit logs for actions affecting ID providers and principal entities (such as `users`, `roles` and `groups`) you have to modify configuration file `com.enonic.xp.security.cfg` and set `auditlog.enabled` option to `true` to enable or `false` to disable logging. Default value is `true`.
```

File: docs/auditLogs.adoc [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      "leaves": "role:system.admin"
+    }
+  }
+}
+----
+
+
+== Application
+
+In order to enable or disable audit logs for `Applications` you have to reference to the `com.enonic.xp.app.cfg` configuration file and set value as `true` to enable and as `false` to disable for the `auditlog.enabled` option. Default value is `true`.
```
Comment (alansemenov) [inline docs/auditLogs.adoc:107]:
```suggestion
In order to enable or disable audit logs for application actions you have to modify configuration file `com.enonic.xp.app.cfg` and set `auditlog.enabled` option to `true` to enable or `false` to disable logging. Default value is `true`.
```

File: docs/auditLogs.adoc [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+=== XP registers the following actions for applications:
+
+* `system.application.start` - an application has been started
+* `system.application.stop` - an application has been stopped
+* `system.application.install` - an application has been installed
+* `system.application.uninstall` - an application has been uninstalled
+
+== Content
+
+In order to enable or disable audit logs for `Content` you have to reference to the `com.enonic.xp.content.cfg` configuration file and set value as `true` to enable and as `false` to disable for the `auditlog.enabled` option. Default value is `true`.
```
Comment (alansemenov) [inline docs/auditLogs.adoc:125]:
```suggestion
In order to enable or disable audit logs for content actions you have to modify configuration file `com.enonic.xp.content.cfg` and set `auditlog.enabled` option to `true` to enable or `false` to disable logging. Default value is `true`.
```

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: docs/auditLogs.adoc (hunk 3 above)
Comment (anatol-sialitski) [reply docs/auditLogs.adoc]: done

File: docs/auditLogs.adoc (hunk 4 above)
Comment (anatol-sialitski) [reply docs/auditLogs.adoc]: done

File: docs/auditLogs.adoc (hunk 5 above)
Comment (anatol-sialitski) [reply docs/auditLogs.adoc]: done

File: docs/auditLogs.adoc (hunk 2 above)
Comment (anatol-sialitski) [reply docs/auditLogs.adoc]: done

**Review by alansemenov — APPROVED** [review]

**Pull request merged**